#446 broke npm state=latest for missing packages

##### Issue Type:
- Bug Report (`npm` module)

##### Ansible Version:
Running against devel:
```console
$ ansible --version
ansible 2.1.0 (devel be5488cb60) last updated 2015/12/15 09:36:59 (GMT -400)
  lib/ansible/modules/core: (devel 6b13da738b) last updated 2015/12/15 09:38:18 (GMT -400)
  lib/ansible/modules/extras: (devel f3251de29c) last updated 2015/12/15 09:38:42 (GMT -400)
  config file = /home/tomxtobin/.ansible.cfg
  configured module search path = Default w/o overrides
```

##### Ansible Configuration:
```ini
[defaults]
hostfile = ~/ansible/hosts
nocows = 1
```

##### Environment:

N/A (but it's Arch Linux)

##### Summary:

It looks like PR #446 broke `npm: name=foo state=latest` for a missing package `foo` (i.e., `foo` isn't present on the system yet).

Suggested fix: for `state == 'latest'`, actually differentiate between the result of checking `len(missing)` and `len(outdated)` to see whether the package is installed or not, and run either `npm.install()` or `npm.update()` as appropriate.

##### Steps To Reproduce:

Let's use the `gulp` package as an example.

On a system that doesn't already have `gulp` installed globally:
```console
$ ansible example-host -m command -a 'npm ls -g -depth 0 gulp' 
example-host | FAILED | rc=1 >>
/usr/lib
└── (empty)npm ERR! code 1
```
```console
$ ansible example-host -m command -a 'gulp --version' 
example-host | FAILED | rc=2 >>
[Errno 2] No such file or directory
```

Run a task against such system to install `gulp` globally:
```console
$ ansible example-host -m npm -a 'name=gulp state=latest global=yes'
example-host | SUCCESS => {
    "changed": true
}
```

##### Expected Results:

The module (`gulp`, above) actually gets installed on the system(s) I'm running that task against.

Against such a system, I can run something like:
```console
$ ansible example-host -m command -a 'npm ls -g -depth 0 gulp' 
example-host | SUCCESS | rc=0 >>
/usr/lib
└── gulp@3.9.0 
```
```console
$ ansible example-host -m command -a 'gulp --version'
example-host | SUCCESS | rc=0 >>
[15:24:28] CLI version 3.9.0
```
(Assuming the latest version of `gulp` happened to be `3.9.0`.)

##### Actual Results:

Ansible claims it succeeds in running the task, but it doesn't actually install `gulp` on the system(s) in question.

On such a system:
```console
$ ansible example-host -m command -a 'npm ls -g -depth 0 gulp' 
example-host | FAILED | rc=1 >>
/usr/lib
└── (empty)npm ERR! code 1
```
```console
$ ansible example-host -m command -a 'gulp --version' 
example-host | FAILED | rc=2 >>
[Errno 2] No such file or directory
```
You can actually keep re-running the task over and over, and Ansible will keep claiming to successfully install `gulp`:
```console
$ ansible example-host -m npm -a 'name=gulp state=latest global=yes'
example-host | SUCCESS => {
    "changed": true
}
$ ansible example-host -m npm -a 'name=gulp state=latest global=yes'
example-host | SUCCESS => {
    "changed": true
}
```

--- a/packaging/language/npm.py
+++ b/packaging/language/npm.py
@@ -252,7 +252,10 @@ def main():
     elif state == 'latest':
         installed, missing = npm.list()
         outdated = npm.list_outdated()
-        if len(missing) or len(outdated):
+        if len(missing):
+            changed = True
+            npm.install()
+        if len(outdated):
             changed = True
             npm.update()
     else: #absent

